sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -0,0 +1,187 @@
+         making requests for partition properties using the Event Hub client.
+         It is set to `False` by default.
+        :param initial_event_position: Start receiving from this initial_event_position
+         if there isn't checkpoint data for a partition. Use the checkpoint data if there it's available. This can be a
+         a dict with partition id as the key and position as the value for individual partitions, or a single
+         EventPosition instance for all partitions.
+        :param on_error: The callback function that is called when there is error during receiving events.
+        :param on_partition_initialize: The callback function that is called right before
+         a partition is initialized.
+        :param on_partition_close: The callback function that is called when this receive method finishes receiving
+         from a partition.
+        :rtype: None
+
+        Example:
+            .. literalinclude:: ../samples/test_examples_eventhub_async.py
+                :start-after: [START eventhub_consumer_client_receive_async]
+                :end-before: [END eventhub_consumer_client_receive_async]
+                :language: python
+                :dedent: 4
+                :caption: Receive events from the EventHub.
+        """
+        async with self._lock:
+            error = None
+            if (consumer_group, '-1') in self._event_processors:
+                error = ValueError("This consumer client is already receiving events from all partitions for"
+                                 " consumer group {}. "
+                                 "Shouldn't receive from any other partitions again".format(consumer_group))
+            elif partition_id is None and any(x[0] == consumer_group for x in self._event_processors):
+                error = ValueError("This consumer client is already receiving events for consumer group {}. "
+                                 "Shouldn't receive from all partitions again".format(consumer_group))
+            elif (consumer_group, partition_id) in self._event_processors:
+                error = ValueError("This consumer is already receiving events from partition {} for consumer group {}. "
+                                 "Shouldn't receive from it again.".format(partition_id, consumer_group))
+            if error:
+                log.warning(error)
+                raise error
+
+            event_processor = EventProcessor(
+                self, consumer_group, on_event,
+                partition_id=partition_id,
+                partition_manager=self._partition_manager,
+                error_handler=on_error,
+                partition_initialize_handler=on_partition_initialize,
+                partition_close_handler=on_partition_close,
+                initial_event_position=initial_event_position or EventPosition("-1"),
+                polling_interval=self._load_balancing_interval,
```

Comment:
Why call it `polling_interval` in `EventProcessor` and `load_balancing_interval` in the public API surface? It is strongly preferred to not change the name for the same context as doing so causes confusion down the line.

Reply:
I agree

Reply:
Not blocking for preview5, solving